src/adapters/kafka/index.ts

```diff
@@ -65,133 +41,22 @@ class KafkaAdapter extends Adapter {
-      const securityRequirements = (this.AsyncAPIServer.security() || []).map(sec => {
-        const secName = Object.keys(sec.json())[0]
-        return this.parsedAsyncAPI.components().securityScheme(secName)
-      })
-      const userAndPasswordSecurityReq = securityRequirements.find(sec => sec.type() === 'userPassword')
-      const X509SecurityReq = securityRequirements.find(sec => sec.type() === 'X509')
-      const url = new URL(this.AsyncAPIServer.url())
-
-      const certsConfig = process.env.GLEE_SERVER_CERTS?.split(',').map(t => t.split(':'))
-      const certs = certsConfig?.filter(tuple => tuple[0] === this.serverName)?.map(t => fs.readFileSync(t[1])) // eslint-disable-line security/detect-non-literal-fs-filename
 
-      // this.kafka = consumer.connect({
-      //   host: url.host,
-      //   port: url.port || (url.protocol === 'kafka:' ? 1883 : 8883),
-      //   protocol: url.protocol.substr(0, url.protocol.length - 1),
-      //   clientId: serverBinding && serverBinding.clientId,
-      //   clean: serverBinding && serverBinding.cleanSession,
-      //   will: serverBinding && serverBinding.will && {
-      //     topic: serverBinding && serverBinding.lastWill && serverBinding.lastWill.topic ? serverBinding.lastWill.topic : undefined,
-      //     qos: serverBinding && serverBinding.lastWill && serverBinding.lastWill.qos ? serverBinding.lastWill.qos : undefined,
-      //     payload: serverBinding && serverBinding.lastWill && serverBinding.lastWill.message ? serverBinding.lastWill.message : undefined,
-      //     retain: serverBinding && serverBinding.lastWill && serverBinding.lastWill.retain ? serverBinding.lastWill.retain : undefined,
-      //   },
-      //   keepalive: serverBinding && serverBinding.keepAlive,
-      //   username: userAndPasswordSecurityReq ? process.env.GLEE_USERNAME : undefined,
-      //   password: userAndPasswordSecurityReq ? process.env.GLEE_PASSWORD : undefined,
-      //   ca: X509SecurityReq ? certs : undefined,
-      // })
-
-      this.kafka.on('connect', () => {
-        if (Array.isArray(subscribedChannels)) {
-          subscribedChannels.forEach((topic) => {
-            const operation = this.parsedAsyncAPI.channel(channel).publish()
-            const binding = operation.binding('kafka')
-            this.kafka.subscribe(channel, {
-              qos: binding && binding.qos ? binding.qos : 0,
-            })
-          })
-        }
+    const run = async () => {
+      await consumer.connect() 
+      const subscribedChannels = this.getSubscribedChannels()
+      await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
+      await consumer.run({
+        eachMessage: async ({ topic, partition, message }) => {
+          const prefix = `${topic}[${partition} | ${message.offset}] / ${message.timestamp}`
+          console.log(`- ${prefix} ${message.key}#${message.value}`)
+        },
       })
+    }
```

Comment:
You don't need to wrap this code inside a `run` function, do you? It should work just fine without it.